### ISSUE TITLE
coap_get_session_client_psk seems to get an incorrect max_identity_len

### ISSUE THREAD

**Issue by vdaele** [obgm/libcoap#493]
I'm using the latest development branch with tinydtls. The length of my psk key is 16, the length of my identity is 32 (I'm authenticating aginst an IKEA gateway)

Autenthication fails in net.c/coap_get_session_client_psk in the following test

```
session->psk_identity->length <= max_identity_len
```
since the max_identity_len parameter has length 16 at this point (and my identity has length 32).

This is the call stack at that point
![Screenshot from 2020-04-28 19-08-13](https://user-images.githubusercontent.com/10400082/80517372-4638ba80-8985-11ea-9a2e-bfcde3eaecaa.png)

**Comment by mrdeep1** [comment]
@vdaele Thank you for raising this.  Fix is in #495 which is waiting for a merge.  Please check this out an confirm that it fixes your issue.

**Comment by vdaele** [comment]
Just a question on the process: do I verify it after it has been merged or do you wait on my confirmation before merging it?

**Comment by mrdeep1** [comment]
If you are able to check the fix out ahead of the merge that would be great as there is a backlog of merges at present.

**Comment by vdaele** [comment]
I've validated the fix and can confirm that this indeed works fine now!

**Comment by mrdeep1** [comment]
@vdaele Now that #495 has been merged into the develop branch, are you happy for this to be closed off?
If so, please close it.

**Comment by vdaele** [comment]
Sure! Feel free to close this!
I won't have time in the near future to validate it myself though
